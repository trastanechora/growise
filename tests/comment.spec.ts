let jsData = {
  id: 18772478,
  user: "thecosas",
  time: 1545943367,
  time_ago: "19 hours ago",
  type: "comment",
  content: "<p>Test</p>",
  comments: [],
  comments_count: 0,
  level: 0,
  url: "item?id=18772478"
}

import { shallowMount, RouterLinkStub } from "@vue/test-utils"
import Comment from "@/components/comment.vue"
// import { timeAgo } from "../plugins/filters"

describe("Comment", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Comment, {
      propsData: {
        comment: jsData
      },
      stubs: {
        RouterLink: RouterLinkStub
      },
      filters: {
        timeAgo(value) {
          return value
        }
      }
    })
  })

  test("snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  // it's also easy to check for the existence of elements

  it("renders Comment", () => {
    //console.log(wrapper.html())
    expect(wrapper.find("div.text *").html()).toBe(jsData.content)
  })
})
