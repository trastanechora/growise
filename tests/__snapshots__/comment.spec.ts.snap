// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment snapshot 1`] = `
<li class="comment">
  <div class="by"><a>thecosas</a>
    1545943367 ago
  </div>
  <div class="text">
    <p>Test</p>
  </div>
  <!---->
  <ul class="comment-children"></ul>
</li>
`;
